src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Wait, I've forgotten: is change_attr supposed to increment on inode changes, or on any change to the file (eg, a data overwrite)?

Do we need to bump the change_attr for every client that incremented it, instead of just taking the ceiling? If it's covered by FILE_EXCL then nobody else should be able to read it without getting an up-to-date version from the MDS, but they also shouldn't be able to *set* it except by talking to the MDS. That doesn't seem to be what's happening here with the setattr code.